index/ask/ask.go

```diff
@@ -2,6 +2,7 @@ package ask
 
```

Comment:
Mostly:
- better names
- refactoring
- bugfixes
- persist index state using `cbor` in a real datastore.